### ISSUE TITLE
Implement the `Hash` trait for `Tn` structs

### ISSUE THREAD

**Issue by tbarusseau** [s3bk/tuple#4]
I just discovered this crate and wanted to integrate it into my Advent of Code solutions, but was met with this error:

```
error[E0277]: the trait bound `T2<usize, usize>: Hash` is not satisfied
    --> src\y2023\day11.rs:33:10
     |
33   |         .collect()
     |          ^^^^^^^ the trait `Hash` is not implemented for `T2<usize, usize>`
     |
     = help: the trait `FromIterator<T>` is implemented for `HashSet<T, S>`
     = note: required for `HashSet<T2<usize, usize>>` to implement `FromIterator<T2<usize, usize>>`
note: required by a bound in `collect`
    --> C:\Users\thoma\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\iter\traits\iterator.rs:2050:19
     |
2050 |     fn collect<B: FromIterator<Self::Item>>(self) -> B
     |                   ^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::collect`

For more information about this error, try `rustc --explain E0277`.
```

Is there a technical limitation that prevents this crate from implementing the `Hash` trait when the contained types already implement `Hash`?

Thanks in advance!

**Comment by s3bk** [comment]
Hello, absolutely not.
I forgot to add the impl.

**Comment by tbarusseau** [comment]
That's great to hear!

Your crate makes working with tuples much better. Thanks a lot for your open-source work!

**Comment by s3bk** [comment]
I published 0.5.2 which has the Hash impl.

**Comment by tbarusseau** [comment]
Thanks a lot ❤️
